richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -304,20 +304,37 @@ public class RichTextState internal constructor(
         textRange: TextRange,
         text: String
     ) {
-        require(textRange.min >= 0) {
-            "The start index must be non-negative."
-        }
+        // Ensure that the start and end indices are within bounds
+        val start = textRange.min.coerceAtLeast(0) // Ensure the start is non-negative
+        val end = textRange.max.coerceAtMost(textFieldValue.text.length) // Ensure the end does not exceed text length
 
-        require(textRange.max <= textFieldValue.text.length) {
-            "The end index must be within the text bounds. " +
-                    "The text length is ${textFieldValue.text.length}, " +
-                    "but the end index is ${textRange.max}."
+        require(start <= end) {
+            "The start index ($start) must be less than or equal to the end index ($end)."
         }
 
-        removeTextRange(textRange)
```

Comment:
After fixing the issue with removing characters, it should work fine. I prefer to keep using removeTextRange + addTextAfterSelection to avoid code duplication. It's not causing any issues since removeTextRange will change the selection to textRange.min.